### ISSUE TITLE
Different results between CI / Theme Check Report and Shopify CLI

### ISSUE THREAD

**Issue by rossb** [Shopify/theme-check-action#49]
Hi,

I'm finding that when theme-check-action runs on PR it fails with different errors than I reported locally with `shopify theme check`. The local lint has no errors, the CI lint has many (and some don't make clear sense). Also even trying different local configs such as `extends:  - theme-check:all` doesn't show the same errors the CI reports.

I assume this must either be a configuration issue, or different versions used by local CLI and CI issue?

My `.theme-check.yml` in the repo looks like this:

```
# Disabled rules
MatchingTranslations:
  enabled: false
VariableName:
  enabled: false
TemplateLength:
  enabled: false

# Enabled with mods
NestedSnippet:
  enabled: true
  max_nesting_level: 3
```

Do I need to point to this config manually from `.github/workflows/ci.yml`? (this is where the workflow is, based on Dawn) I don't see any documented options for config, though I did try pointing it via `flags` and it made no difference to CI reports.

An example issue shown by the CI (there is no invalid token at the start of this JSON, so I don't know what it refers to):

![Image](https://github.com/user-attachments/assets/ed9eb9a8-65c4-4ce9-b51f-0fd6c9f0cd97)

I'm also getting a lot of "[SchemaJsonFormat] JSON formatting could be improved" where I don't see any issues with the formatting, and the local Theme Check doesn't report any issues.

I'm using Shopify CLi 3.68.1 locally, and my workflow is as below (it's the latest default Dawn version):

```
  theme-check:
    name: Theme Check
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - name: Theme Check
        uses: shopify/theme-check-action@v1
        with:
          token: ${{ github.token }}
```

Also I can see the config used in the CI report, but even copying this config locally into the repo (it's very different to above) and pointing the CLI to it locally doesn't give me the same results (it still shows 0 errors locally).

Any help would be appreciated.

**Comment by jamesmengo** [comment]
Can you try updating the version of `theme-check-action` you're using to the latest version to see if that resolves your issue?

**Comment by rossb** [comment]
Thanks @jamesmengo, this worked and now makes sense, `@v1` is the old theme check and the version in latest CLI matches `shopify/theme-check-action@v2`. My results match perfectly :)

It could be worth adding a note to README for others setting it up first time on latest Dawn, latest CLI, and just assume both will be synced up on versions (I'm guessing backwards compatibility has kept Dawn at @v1).
